flow/environment.py

```diff
@@ -41,6 +42,17 @@
 logger = logging.getLogger(__name__)
 
 
+@lru_cache(maxsize=1)
+def _cached_fqdn():
+    """Returns the fully qualified domain name.
+
+    If called for the first time in a session, call socket.getfqdn
+    and cache the value. If called again, return the cached value.
+    Solves #339
```

Comment:
We typically do not put issue/PR references in the code, just the changelog. You could instead write:
```suggestion
    This value is cached because fetching the fully qualified domain name can
    be slow on macOS.
```